core-plugins/src/main/java/co/cask/hydrator/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.joiner;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.plugin.PluginConfig;
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.annotation.Nullable;
+
+/**
+ * Config for join plugin.
+ */
+public class JoinerConfig extends PluginConfig {
+  private static final String NUM_PARTITIONS_DESC = "Number of partitions to use when joining. " +
+    "If not specified, the execution framework will decide how many to use.";
+  private static final String JOIN_KEY_DESC = "List of join keys to perform join operation. The list is " +
+    "separated by join condition. Join key from each input stage will be prefixed with '<stageName>.' And the " +
+    "relation among join keys from different inputs is represented by '='. For example: customers.customer_id=items" +
+    ".c_id&customers.customer_name=items.c_name will mean the join key is a composite key of customer id and customer" +
+    " name from customers and items input stages and join will be performed on equality of the join keys.";
+  private static final String SELECTED_FIELDS = "Map which represents fields to be selected and " +
```

Comment:
this seems like a pretty difficult format to understand and get correct.  Can we make this simpler for users by supporting a sql-like syntax where its a list, but people can say `as xyz` to rename the field to `xyz`? So something like `A.id as user_id, B.id as product_id`